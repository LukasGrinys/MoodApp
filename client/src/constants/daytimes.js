export const daytimes = {
    morning: 'Morning',
    afternoon: 'Afternoon',
    evening: 'Evening',
    night: 'Night'
}