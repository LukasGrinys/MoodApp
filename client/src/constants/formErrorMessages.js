export const formErrorMessages = {
    required : 'Required field',
    email: 'Enter a proper e-mail address'
}