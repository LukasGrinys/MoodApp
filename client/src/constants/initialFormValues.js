export const initialFormValues = {
    email: '',
    password: '',
    firstName: '',
    lastName: '',
    confirmPassword: '',
    currentPassword: ''
}