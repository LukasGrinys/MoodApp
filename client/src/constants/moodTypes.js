const moodTypes = [
    {
        above: 7.5,
        below: 10,
        text: 'Great'
    },
    {
        above: 5,
        below: 7.5,
        text: 'Good'
    },
    {
        above: 3,
        below: 5,
        text: 'Average'
    },
    {
        above: 0,
        below: 3,
        text: 'Bad'
    }
];

export default moodTypes;