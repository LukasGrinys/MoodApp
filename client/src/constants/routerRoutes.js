export const routerRoutes = {
    home: '/',
    login: '/login',
    logout: '/logout',
    dashboard: '/dashboard',
    addNewLog: '/addlog',
    register: '/register',
    settings: '/settings'
}