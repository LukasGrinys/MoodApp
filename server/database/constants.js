const COLLECTION_NAMES = {
    users : 'users',
    logs : 'logs'
}

module.exports = {
    COLLECTION_NAMES
}