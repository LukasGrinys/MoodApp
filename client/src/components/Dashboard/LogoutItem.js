import React from 'react';
import { Link } from 'react-router-dom';
import { useTheme } from '../../contexts/ThemeContext';
import { routerRoutes } from '../../constants/routerRoutes';
import parentClasses from './Dashboard.module.scss';
import styles from './LogoutItem.module.scss';
import classNames from 'classnames';
import FontAwesome from 'react-fontawesome';

const LogoutItem = () => {
    const darkTheme = useTheme();

    return (
        <div className={classNames(
            parentClasses.gridItem,
            parentClasses.shrink
            )}>
            <h3 className={parentClasses.gridItemHeader}>Log out</h3>
            <div className={parentClasses.flexCenter}>
                <Link to={routerRoutes.logout}>
                    <div className={styles.logoutIconWrapper}> 
                        <FontAwesome name="sign-out" className={classNames(
                            styles.logoutIcon,
                            darkTheme && styles.dark
                        )}/>
                    </div>
                </Link>
            </div>
        </div>
    )
};

export default LogoutItem;