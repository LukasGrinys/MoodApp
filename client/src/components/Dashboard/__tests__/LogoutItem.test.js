import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import LogoutItem from '../LogoutItem';
import { mount } from 'enzyme';
import { routerRoutes } from '../../../constants/routerRoutes';

const wrapper = mount(
    <Router>
        <LogoutItem/>
    </Router>
)

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Has logout icon', () => {
    const icon = wrapper.find('FontAwesome');

    expect(icon.length).toBe(1);
    expect(icon.props()).toEqual({
        className: 'logoutIcon',
        name: 'sign-out'
    })
});

test('Has a link to logout page', () => {
    const link = wrapper.find('Link');
    
    expect(link.length).toBe(1);
    expect(link.prop('to')).toBe(routerRoutes.logout);
})