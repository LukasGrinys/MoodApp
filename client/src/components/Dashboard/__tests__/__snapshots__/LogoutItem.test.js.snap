// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LogoutItem>
      <div
        className="gridItem shrink"
      >
        <h3
          className="gridItemHeader"
        >
          Log out
        </h3>
        <div
          className="flexCenter"
        >
          <Link
            to="/logout"
          >
            <LinkAnchor
              href="/logout"
              navigate={[Function]}
            >
              <a
                href="/logout"
                onClick={[Function]}
              >
                <div
                  className="logoutIconWrapper"
                >
                  <FontAwesome
                    className="logoutIcon"
                    name="sign-out"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-sign-out logoutIcon"
                    />
                  </FontAwesome>
                </div>
              </a>
            </LinkAnchor>
          </Link>
        </div>
      </div>
    </LogoutItem>
  </Router>
</BrowserRouter>
`;
